lib/Dialect/Triton/IR/Types.cpp

```diff
@@ -157,7 +157,15 @@ Type getPointerTypeSameShape(Type type) {
   }
 }
 
-Type getPointerType(Type type) { return PointerType::get(type, 1); }
+Type getPointerType(Type type, int addressSpace) {
+  return PointerType::get(type, addressSpace);
+}
+
+int getAddressSpace(Type type) {
+  if (auto ptrType = dyn_cast<PointerType>(type))
+    return ptrType.getAddressSpace();
+  return 1;
+}
```

Comment:
Why would the pointer be anything else than address space 1 for blocked pointer? The op is meant to only support global memory pointers and it is likely to break in other cases

Reply:
we use block pointer for global memory and share local memory for Intel GPU.
it works well in our local branch. 
so we'd like to upstream this and it do not break upstream code since 1 is still the default addrspace.